packages/neon/neon_notes/lib/src/blocs/note.dart

```diff
@@ -63,10 +66,14 @@ class _NotesNoteBloc extends InteractiveBloc implements NotesNoteBloc {
         final response = await call(etag);
         emitNote(response.body);
         await notesBloc.refresh();
-      } catch (e, s) {
-        debugPrint(e.toString());
-        debugPrint(s.toString());
-        addError(e);
+      } on http.ClientException catch (error, stackTrace) {
```

Comment:
I think we should always catch all exceptions. If a different exception is triggered then the user will get no feedback about it which is not the point of error handling.

Reply:
Where should a different exception come from?
`notesBloc.refresh` should handle the errors itself (in this case through the request manager) while emitNote shouldn't ever throw.
This leaves us with `call` which indicated by the return type is a dynamite request. Dynamite requests only throw `http.ClientException` (it does the error handling for us and propagates them as a ClientException).

Shure we could have a callback that does some custom stuff, throws custom objects and returns a DynamiteResponse but then the code would be awful anyways.

Reply:
I mean it could also be a TimeoutException or SocketException. There are many different errors that can appear.

Reply:
Assuming that the call is always a normal nextcloud request (`account.client.news.getSomething()`) where would such an exception come from?

Reply:
Ok I had a closer look and it seems like http package wraps all Exceptions using it's ClientException.
Then this will not be an issue, but I still think it is better to just handle all Exceptions. We gain nothing by only catching some (only worse UX) and it is much more likely that we introduce bugs when refactoring, in case a different Exception could be thrown at some point.

Reply:
Just so we are on the same page. I changed many catch statements where no on clause was specified to only catch the appropriate object.

Do you want to change it everywhere to catch arbitrary `Exception`s and `Error`s or just in this particular case?

Reply:
Everywhere please. So we just catch `Exception` everywhere, right?

Reply:
I disagree. We should only catch the exceptions we expect to happen so unexpected issues do not get unnoticed. Error handling should happen where the error is expected and in this case only an ClientException.

imagine a scenario like:
```dart
void main() async {
  final subject = BehaviorSubject.seeded('seedValue');

  try {
    await subject.close();
    subject.add('event');
  } catch (_) {
    print('nothing wrong here');
  }
}
```

We wouldn't notice the `Bad state: Cannot add new events after calling close` (I know that this example actually throws a StateError which is not a subtype of Exception but you get my point).

My long term goal which we are fairly close to after this PR is to enable the [avoid_catches_without_on_clauses](https://dart.dev/tools/linter-rules/avoid_catches_without_on_clauses) lint rule. 
It basically says the same thing:
> Using catch clauses without on clauses make your code prone to encountering unexpected errors that won't be thrown (and thus will go unnoticed).

Also the docs on the Exception class itself (https://api.dart.dev/stable/3.3.0/dart-core/Exception-class.html) mention:
> Creating instances of Exception directly with Exception("message") is discouraged in library code since it doesn't give users a precise type they can catch.


In most literature about error handling one would find the advice "do not catch generic errors" and I'm hesitant in ignoring this advice.

Reply:
I get the point of doing that, but I still think the UX will suffer. We can go this route, but I'm sure at some point we will have bug reports saying that something doesn't work but there is also no error displayed (would only appear in the logs).

Reply:
for this I'd like to use `FlutterError.onError` and the `PlatformDispatcher.instance.onError` callbacks (the same used by firebase_crashlytics, sentry and probably other similar services).
They are built for that exact usecase:
https://docs.flutter.dev/testing/errors
https://docs.flutter.dev/cookbook/maintenance/error-reporting (probably not so important for our usecase)

Reply:
Right, let's do that

Reply:
I'll open an issue and add it to the release milestone. Is that ok?

Reply:
I feel like that needs to be solved before we can merge this, otherwise we will miss some of the exceptions (in case there is a programming error somewhere).

Reply:
Ok so we will do the global error handling later.